doc/developer/design/20230223_stabilize_with_mutually_recursive.md

```diff
@@ -0,0 +1,297 @@
+`literal_constraints`        | ...      | ...
+`literal_lifting`            | ...      | ...
+`map_fusion`                 | xs       | trivial
+`negate_fusion`              | xs       | trivial
+`non_null_requirements`      | ...      | ...
+`non_nullable`               | ...      | ...
+`normalize`                  | ...      | ...
+`normalize_lets`             | &check;  | MaterializeInc/materialize#16665
+`predicate_pushdown`         | ...      | ...
+`project_fusion`             | xs       | trivial
+`projection_extraction`      | ...      | ...
+`projection_lifting`         | ...      | ...
+`projection_pushdown`        | ...      | ...
+`reduce_elision`             | ...      | ...
+`reduce_fusion`              | xs       | trivial
+`reduction_pushdown`         | ...      | ...
+`redundant_join`             | ...      | ...
+`relation_cse`               | xl       | hard?
+`semijoin_idempotence`       | ...      | ...
+`threshold_elision`          | ...      | ...
+`topk_elision`               | ...      | ...
+`topk_fusion`                | xs       | trivial
+`union`                      | xs       | trivial
+`union_branch_cancellation`  | ...      | ...
+`union_negate`               | xs       | trivial
+
+## Generalization of LIR rendering
+
+This should be handled by MaterializeInc/materialize#17705.
+We might have to add more tests for that (see [Testing and observability](#testing-and-observability)).
+
+# Rollout
+[rollout]: #rollout
+<!--
+Describe what steps are necessary to enable this feature for users.
+How do we validate that the feature performs as expected? What monitoring and observability does it require?
+-->
+The WMR feature is currently only enabled in `--unsafe-mode`.
+As part of the enclosing epic, we will introduce a dedicated `with_mutually_recursive` feature flag.
+The feature will be first made available on all `staging` and gradually rolled out to selected `production` environments.
+The following aspects need special attention:
+1. Queries producing wrong results (discussed in [Testing and observability](testing-and-observability)).
+1. Queries that do not terminate. This is tricky because some queries might be divergent because of a bad query definition (a user error) instead of an optimization or interpretation bug (a system error). A related issue to track this is #16800.
+
+To validate (1), I suggest to:
+
+* Ask [the DevEx team](https://github.com/orgs/MaterializeInc/teams/devex) to deploy WMR materialized views on their canary environments.
+* Use the internal observability metrics as early adopters for WMR.
+
+Validating (2) is [an open question](#unresolved-questions).
+
+## Testing and observability
+[testing-and-observability]: #testing-and-observability
+<!--
+Testability and explainability are top-tier design concerns!
+Describe how you will test and roll out the implementation.
+When the deliverable is a refactoring, the existing tests may be sufficient.
```

Comment:
This assumes we already have good testing coverage for it. Is that always the case?

Reply:
No.